### ISSUE TITLE
Cats - Extension Methods For Future/Task Conversion

### ISSUE THREAD

**Issue by ChristopherDavenport** [http4s/http4s#913]
Rather than consistently wrapping entire function calls we could embed implicit classes within the org.http4s.util.Task implementation. 

Something like
```scala
object task extends TaskFunctions {
  implicit class TaskToFutureConverter[A](task: Task[A]){
    def unsafeToFuture : Future[A] = unsafeTaskToFuture[A](task)
  }
  implicit class FutureToTaskConverter[A](future: => Future[A])(implicit ec: ExecutionContext) {
    def toTask : Task[A] = futureToTask(future)(ec)
  }
}
```

This would allow us to interact on the edge of blaze/http4s much easier. The only major question is if this is required, why isn't it directly in fs2? However this would in the implementation of the writeEntityBody.

```scala
  def writeEntityBody(p: EntityBody): Task[Boolean] = {
    val writeBody : Task[Unit] = (p to writeSink).run
    val writeBodyEnd : Task[Boolean] = writeEnd(Chunk.empty).toTask
    writeBody >> writeBodyEnd
  }
  
  private val writeSink: Sink[Task, Byte] = { s =>
    val writeStream : Stream[Task, Unit] = s.chunks.evalMap[Task, Task, Unit](writeBodyChunk(_ , false).toTask)
    val errorStream : Throwable => Stream[Task, Unit] = e => Stream.eval(exceptionFlush().toTask).flatMap{_ => fail(e)}
    writeStream.onError(errorStream)
  }
```

**Comment by chuwy** [comment]
Sorry if I misinterpret, but how is this different from [`unsafeRunAsyncFuture`](https://github.com/functional-streams-for-scala/fs2/blob/series/1.0/core/shared/src/main/scala/fs2/Task.scala#L157)?

**Comment by ChristopherDavenport** [comment]
I would say that we can scratch that one, as it replaces the current implementation of unsafeTaskToFuture exactly, and it is on Task.

Infact on further investigation, we may be able scrap this whole package in favor of `unsafeRunAsyncFuture` and `Task.fromFuture` which allows us to move the blaze abstractions to the edge.

**Comment by rossabaker** [comment]
I agree.  We needed this for scalaz Task, but it's built into fs2 Task.

If anyone needs this functionality for Scalaz, we use [delorean](https://github.com/Verizon/delorean) at work.  I don't want to add the dependency to our scalaz branch for something so slight, but that's the way to do it in apps.

**Comment by ChristopherDavenport** [comment]
Alright, I'll see if I can do a refactor pr to phase out the use of the util.Task package. 

I'm not avoiding that bug that I can't seem to find... :pager:

**Comment by ChristopherDavenport** [comment]
Closed By #916
